### PR TITLE
Make test concurrency work

### PR DESCRIPTION
**Pull request by kat1906**
Concurrency still doesn't work
- The tests should now all run synchronously
- I have commented out the nock afterEach hooks
- Now throws error if the type of Q doesn't have any tests written for it

### CODE REVIEW

File: app/section/helpers/transformSectionData.js [hunk 1]
```diff
@@ -10,10 +10,15 @@ const arrayTransformation = (questionValue) => {
   return [questionValue];
 };
 
+const radioButtonTransformation = questionValue => (questionValue || null);
```
Comment (alibombaye) [inline app/section/helpers/transformSectionData.js]: I would name this function to something like 'emptyValueTransformer' or something along them lines. We could use this for something else?

Comment (kat1906) [reply app/section/helpers/transformSectionData.js]: it was stringTranformation until API got involved ... 😄

File: app/test-utils/testFunctions/commonComponentsTests.js [hunk 2]
```diff
@@ -49,18 +49,18 @@ const submitButtonTest = ({ pageSetup, sectionManifest }) => {
...
   sectionManifest,
   dashboardId,
 }) => {
   if (sectionManifest.successfulSubmitResponsePath) {
-    test(`should go to ${sectionManifest.successfulSubmitResponsePath} when clicking the submit button`, async (t) => {
+    await test(`should go to ${sectionManifest.successfulSubmitResponsePath} when clicking the submit button`, async (t) => {
       await pageSetup({ t, responseBody: data });
       nock(apiLocalhost)
-        .put(`${apiPath}/sections/${sectionId}`)
+        .put(`${apiPath}/sections/${sectionId}`, data)
```
Comment (alibombaye) [inline app/test-utils/testFunctions/commonComponentsTests.js]: the request body is the same as the response body?

Comment (kat1906) [reply app/test-utils/testFunctions/commonComponentsTests.js]: errrrrr yeah this doesn't look right

File: app/test-utils/testFunctions/questionTests/errorTests.js [hunk 3]
```diff
@@ -129,28 +130,39 @@ const mandatoryErrorTest = ({
...
 }) => {
-  Object.keys(questionData.errorResponse).forEach((errorType) => {
+  const errorPostBody = Object.keys(sectionManifest.questions).reduce((acc, question) => {
+    const questionType = sectionManifest.questions[question].type;
+    if (questionType === 'radiobutton-options') acc[question] = null;
+    else if (questionType === 'checkbox-options') acc[question] = [];
+    else if (questionType === 'bulletpoint-list') acc[question] = new Array(sectionManifest.questions[question].maxItems).fill('');
+    else acc[question] = '';
+    return acc;
+  }, {});
```
Comment (alibombaye) [inline app/test-utils/testFunctions/questionTests/errorTests.js]: why can't we just pass in from the ui.test the errorPostBody we want to return?

Comment (kat1906) [reply app/test-utils/testFunctions/questionTests/errorTests.js]: I did start doing this but it is repetitive and predictable so why not automate it??

File: app/test-utils/testFunctions/commonComponentsTests.js (hunk 2 above)
Comment (kat1906) [reply app/test-utils/testFunctions/commonComponentsTests.js]: so this is correct as just above we send in the data to populate the component so when submit is pressed again it just resubmits what is there ie the prepopulated data

**Pull request merged**